exercises/anagram/canonical-data.json

```diff
@@ -131,6 +131,15 @@
         "candidates": ["Banana"]
       },
       "expected": []
+    },
+    {
+      "description": "detects mixed-case anagrams",
```

Comment:
This appears to be a repeat of `"detects anagrams case-insensitively"`

Reply:
Oh yes you're right! That way, should I make the `"detects anagrams case-insensitively"` harder?

E.g.:
```
 {
   "description": "detects anagrams case-insensitively",
   "property": "anagrams",
   "input": {
     "subject": "OrChEsTrA",
     "candidates": ["cashregister", "CaRtHoRsE", "radishes"]
   },
   "expected": ["CaRtHoRsE"]
 }
```

Reply:
Why is that harder?  Still seems redundant.

Reply:
It's harder in the sense that there's more mixed-case letters. For e.g.:

```
 {
   "description": "detects simple anagram",
   "property": "anagrams",
   "input": {
     "subject": "ant",
     "candidates": ["tan", "stand", "at"]
    },
    "expected": ["tan"]
 },

 {
   "description": "detects anagram",
   "property": "anagrams",
   "input": {
     "subject": "listen",
     "candidates": ["enlists", "google", "inlets", "banana"]
   },
   "expected": ["inlets"]
 },
```

Both tests that I mentioned above are actually testing for the same thing (lower-cased anagram matches). The length differentiates between the difficulty.

Or do you think it's unnecessary, and I should remove one of the test case above as well? :P

Reply:
The more I think about this the less I know what to do.  I think at the very least `"detects mixed-case anagrams"` could be moved up to be with the other `orchestra` test cases.

@Insti, @petertseng or @ErikSchierboom do you have any thoughts on this?

Maybe nothing should be done, and just merge this PR as is?

Reply:
@rpottsoh following what's been mentioned [here](https://github.com/exercism/problem-specifications/pull/1119#issuecomment-357855859) I should: 
1. Have a new commit to remove the test with description `"detects simple anagram"`, 
1. Remove this test.